components/ActivityForm/index.js

```diff
@@ -23,19 +21,15 @@ export default function ActivityForm({ onSubmit, initialData, isEditMode }) {
 
   return (
     <>
-      <StyledBackLink href={isEditMode ? `/${initialData?.id}` : "/"}>
-        <Icon name="chevronLeft" />
-        {isEditMode ? "Discard changes" : "Back to all Activities"}
-      </StyledBackLink>
       <StyledForm onSubmit={handleSubmit}>
```

Comment:
I just saw that this form has a `display: grid;`. Why did you choose it? Grid's advantage is to create a multi-dimensional layout while flex is perfect for one-dimensional layouts. Since you only have one column try `display: flex; flex-direction: column;` (you won't need justify-content anymore). Another positive effect is that you now use the complete width of the container and the back link on the top is not out of line anymore.